pkg/scheduler/actions/preempt/preempt_test.go

```diff
@@ -18,15 +18,185 @@ package preempt
 
 import (
 	"testing"
+	"time"
 
-	"github.com/kubernetes-sigs/volcano/pkg/scheduler/framework"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
 
-	"github.com/kubernetes-sigs/volcano/pkg/scheduler/plugins/drf"
+	kbv1 "github.com/kubernetes-sigs/volcano/pkg/apis/scheduling/v1alpha1"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/api"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/cache"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/conf"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/framework"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/plugins/conformance"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/plugins/gang"
+	"github.com/kubernetes-sigs/volcano/pkg/scheduler/util"
 )
 
 func TestPreempt(t *testing.T) {
-	framework.RegisterPluginBuilder("drf", drf.New)
+	framework.RegisterPluginBuilder("conformance", conformance.New)
+	framework.RegisterPluginBuilder("gang", gang.New)
 	defer framework.CleanupPluginBuilders()
 
-	// TODO (k82cn): Add UT cases here.
+	tests := []struct {
+		name      string
+		podGroups []*kbv1.PodGroup
+		pods      []*v1.Pod
+		nodes     []*v1.Node
+		queues    []*kbv1.Queue
+		expected  int
+	}{
+		{
+			name: "one Job with two Pods on one node",
+			podGroups: []*kbv1.PodGroup{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pg1",
+						Namespace: "c1",
+					},
+					Spec: kbv1.PodGroupSpec{
+						Queue: "c1",
```

Comment:
nit: The queue name is `c1`  and the namespace is also called `c1`

Reply:
Done.  Have changed queue name to q1